fpga_interchange/pseudo_pip_model.cc

```diff
@@ -378,13 +377,22 @@ void PseudoPipModel::update_site(const Context *ctx, size_t site)
         cell.type = IdString(ctx->wire_lut->cell);
         NPNR_ASSERT(ctx->wire_lut->input_pins.size() == 1);
         cell.lut_cell.pins.push_back(IdString(ctx->wire_lut->input_pins[0]));
-        cell.lut_cell.equation.resize(2);
-        cell.lut_cell.equation.set(0, false);
-        cell.lut_cell.equation.set(1, true);
 
-        // Map LUT input to input wire used by pseudo pip.
-        IdString input_bel_pin(bel_data.ports[bel.input_bel_pin]);
-        cell.cell_bel_pins[IdString(ctx->wire_lut->input_pins[0])].push_back(input_bel_pin);
+        if (bel.input_bel_pin == -1) {
+            // FIXME: currently assume that LUT route-throughs with no input pins are GND drivers as this is all we need
+            // for Nexus/Xilinx where Vcc is readily available and cheap This won't be true for other arches
+            cell.lut_cell.equation.resize(2);
+            cell.lut_cell.equation.set(0, false);
+            cell.lut_cell.equation.set(1, false);
```

Comment:
This allows having constant GND generated by the LUT, is this correct?

Reply:
as it stands yes, see the comment this is the main use case for LUT pseudo-cells with no input pins. supporting anything else will need more work on the schema side, I think

Reply:
All right. As I am dealing with the LUT-thru FASM generation, I wonder how to express and more importantly identify this case. 

For instance, at the moment I am implementing the logic to identify the LUT-thrus by examining whether a LUT bel is traversed entirely in the same physical net, meaning that the LUT was indeed used as a route-thru.

In this case, however, I assume (maybe wrongly) that the net will start from the pseudoCell output, in this case, the LUT output, to reach the various sinks, therefore making the above logic unusable, as the input pins of the LUT appears as unused.

Reply:
I think this special case could be identified as the net being a ground/vcc net on the LUT output, which should force the LUT function for that net to all-zeros or all-ones accordingly.

Reply:
Ah, right, cause the net will be a const net, therefore yes, this case can be actually easily identified. Thanks for the clarification.